src/Common/tests/System/PlatformDetection.cs

```diff
@@ -408,5 +405,33 @@ public static bool IsNonZeroLowerBoundArraySupported
         // System.Security.Cryptography.Xml.XmlDsigXsltTransform.GetOutput() relies on XslCompiledTransform which relies
         // heavily on Reflection.Emit
         public static bool IsXmlDsigXsltTransformSupported => PlatformDetection.IsReflectionEmitSupported;
+
+        public static Range[] FrameworkRanges => new Range[]{
+          new Range(new Version(4, 7, 2500, 0), null, new Version(4, 7, 1)),
+          new Range(new Version(4, 6, 2000, 0), new Version(4, 7, 2090, 0), new Version(4, 7, 0)),
+          new Range(new Version(4, 6, 1500, 0), new Version(4, 6, 1999, 0), new Version(4, 6, 2)),
+          new Range(new Version(4, 6, 1000, 0), new Version(4, 6, 1499, 0), new Version(4, 6, 1)),
+          new Range(new Version(4, 6, 55, 0), new Version(4, 6, 999, 0), new Version(4, 6, 0)),
+          new Range(new Version(4, 0, 30319, 0), new Version(4, 0, 52313, 36313), new Version(4, 5, 2))
```

Comment:
Haha no worries. I can show you the spreadshit on Monday. In example, we shipped 4.5.1 version as 4.0.2300.2267 and 4.5.2 version as 4.0.1019.3927 and as you can see the 4.5.1 is > than 4.5.2 so the if pattern for some of the 4.5.1 will return 4.5.2

The problem is that the incremental version pattern started from 4.6 and before the build and revision numbers where calculated through the date it was built and since we have a lot of shipping/servicing branches before 4.6 there are a lot of builds for each framework <= 4.5.2 which intersect and not always are incremental.